code/modules/projectiles/gun.dm

```diff
@@ -713,6 +713,13 @@ obj/item/weapon/gun/Destroy()
 		to_chat(user, "<span class='notice'>You attach \the [I] to the front of \the [src].</span>")
 		update_icon()
 
+	if(iscrowbar(I))
+		if(bayonet)

```

Comment:
Coding preference, but this could just be one if statement ``if(iscrowbar(I) && bayonet)``